added chakra-ui npm package and palette components

r1g023 opened the pull request:
# Description
- added Palette Header, color (circled) icons and a copy hex button per the Figma guide

Issue # #21 

Describe what you're adding
- added Header and icons per Figma guide
- Added a copy hex values button
- added colored icons
- Adjust header / Text per Figma specifications 

# QA

List out how to test your changes and add a screenshot of what it should look like, when applicable.
![palette](https://github.com/Category-Colors/category-colors/assets/57161327/599d69b7-0463-4f15-b153-588fd0dd89f8)

r1g023 committed:
squashed - removed some button props, updated PalHeader, removed frontWeigh from LandingPage

r1g023 commented:
@zeeberry 
@ilikescience 

I still need to adjust some things and will be creating a second commit sometime later today before our meeting. I added everything within one component and feel like I could have isolated each section into different components so that the PaletteMainCard looks cleaner. 

I tried as best as I could to add the text size, colors, etc.. per the Figma guide but I might need further guidance on that if necessary. 

The task looks almost the same as the Figma file but again, needs a little bit more tweaking which I should be able to do before our meeting tomorrow 9/27 (or later today).

r1g023 commented:
@ilikescience Hi Matt, I was refactoring the main Palette section to have its own components but noticed on the Figma design you created there are 2 things I'm unsure of, see below: 

- leftIcon: IoCopy 
- Colors: gray/200 #E2E8F0 

The IoCopy icon, is that something I'm getting from Chakra, or is it a React icon? For the second one, I'm not sure if I should be adding the gray.200 color to the Copy Hex button or just the text within the button (not including the icon). Whenever I try to add a gray background, the button kinda fades away unless I add a darker color. I might be missing something so that's why I wanted to reach out and ask.

zeeberry commented:
hey @r1g023, super tiny request -- can you update the issue number in the PR description? it should be #21 instead of #34. thanks!

ilikescience commented on src/components/PaletteCopyButton.js:
```diff
@@ -0,0 +1,22 @@
+      color="black"
+      variant="outline"
+      colorScheme="gray"
+      br={"6px"}
```
The `br`, `sx`, and `border` props here are redundant; styling is handled by the `variant`, `colorScheme`, and `size` props.

r1g023 replied:
Done, makes perfect sense, have removed props and the code looks cleaner, thank you!

ilikescience commented on src/components/PaletteCopyButton.js:
```diff
@@ -0,0 +1,22 @@
+      variant="outline"
+      colorScheme="gray"
+      br={"6px"}
+      size={"sm"}
```
should these curly braces be removed?

r1g023 replied:
I removed the br prop and added just 'sm' for size, good catch :)

ilikescience commented on src/components/PaletteCopyButton.js:
```diff
@@ -0,0 +1,22 @@
+      sx={{ padding: "0px 12px 0px 12px", width: "158px" }}
+      border={"1px solid #E2E8F0"}
+    >
+      <CopyIcon fontSize={20} sx={{ marginRight: "10px" }} />
```
The button has a feature where you can pass an icon in as a prop instead of a child. This will handle the `fontSize` and styling props automatically!

r1g023 replied:
Thank you for the feedback, per our previous conversation, I have added a leftIcon props to the button and added the value of "leftIcon={<CopyIcon />}" to it, all set.

r1g023 commented:
I added a few more commits with my changes and squashed them all into 1 single commit, please let me know this is okay to merge to main @zeeberry @ilikescience

ilikescience commented:
(cc @r1g023 ) @zeeberry is it ok if we close this PR and delete the branch? Then we can mark the task done in the project section — seems like it got handled in another branch/PR
